### PR TITLE
Update to 1.2.5. JB#52079

### PR DESCRIPTION
**Pull request by pvuorela**
Updated to new version following the README.sailfishos.

Some notable details:
- Upstream switched to meson, thus mce commit needed here a bit of adjustment on the build system side.
- There was one strange zypp commit for removing the whole pk_backend_upgrade_system() which we were also adjusting. The comments on that seemed nonsense and a similar function got later added back in the main branch. More details on the revert commit.
- Got rid of a bunch of cruft on the rpm packaging. Also squashed all the .spec change commits.
- Minor adjustments on the sailfish readme
- Changed few sailfish commits to have the "sfos:" comment style.
- Jusa's sqlite commit was slightly different merged upstream, assumed that was the more polished version.
- Some minor adjustments to sailfish policy commit for removing unused variable and moving another one on different place to avoid c compiler warning on declaration between code.

Not much testing so far. Maybe the thing I'd be the most cautious is if the upstream zypp backend changes are all ok. Didn't go through those with thought yet.

### CODE REVIEW

**Comment by Thaodan** [comment]
Just wondering with which libzypp version the upstream tested, is our libzypp version still fine?

File: rpm/PackageKit.spec [hunk 1]
```diff
@@ -17,26 +17,15 @@ Requires: shared-mime-info
...
-BuildRequires: pam-devel
 BuildRequires: sqlite-devel
 BuildRequires: polkit-devel >= 0.98
-BuildRequires: libtool
-BuildRequires: perl(XML::Parser)
-BuildRequires: intltool
+BuildRequires: meson
 BuildRequires: gettext
-BuildRequires: libarchive-devel
-BuildRequires: fontconfig-devel
```
Comment (Thaodan) [inline rpm/PackageKit.spec:28]: This was only needed to the gtk backend (just adding why it was removed).

File: rpm/PackageKit.spec [hunk 2]
```diff
@@ -193,7 +156,6 @@ update-mime-database %{_datadir}/mime &> /dev/null || :
 %dir %{_sysconfdir}/PackageKit
 %dir %{_localstatedir}/lib/PackageKit
 %dir %{_localstatedir}/cache/PackageKit
-%dir %{_localstatedir}/cache/PackageKit/downloads
 %dir %{_libdir}/packagekit-backend
 %config %{_sysconfdir}/PackageKit/*.conf
 %config %{_sysconfdir}/dbus-1/system.d/*
```
Comment (Thaodan) [inline rpm/PackageKit.spec:161]: Dbus services should go to `%{_datadir}` as done in upstream, Fedora and already done for Polkit.

**Review by Thaodan — CHANGES_REQUESTED** [review]
Added a minor comment to the spec file.

**Comment by pvuorela** [comment]
> Just wondering with which libzypp version the upstream tested, is our libzypp version still fine?

Weell, the actual upstream probably didn't test any version but rather the people doing contributions so that's probably what is there on some newish opensuse. Then again the changes don't generally seem that big or related to libzypp behavior changes.

File: rpm/PackageKit.spec (hunk 2 above)
Comment (pvuorela) [reply rpm/PackageKit.spec:161]: There are both, it's the same as Fedora, there is no changes and it should be up to the upstream to make directory changes.

Comment (Thaodan) [reply rpm/PackageKit.spec:161]: Upstream defaults to datadir too: https://github.com/PackageKit/PackageKit/blob/b54014f6e7650ed28f16714c95fd7f84d65bc613/data/meson.build#L25.

Comment (pvuorela) [reply rpm/PackageKit.spec:161]: And upstream defaults sysconfdir: https://github.com/PackageKit/PackageKit/blob/b54014f6e7650ed28f16714c95fd7f84d65bc613/data/meson.build#L12

As I said, it uses both.

Comment (Thaodan) [reply rpm/PackageKit.spec:161]: > And upstream defaults sysconfdir: https://github.com/PackageKit/PackageKit/blob/b54014f6e7650ed28f16714c95fd7f84d65bc613/data/meson.build#L12

that's for the config, I'm talking about the dbus sevice.

Comment (pvuorela) [reply rpm/PackageKit.spec:161]: Sounds like you didn't test how this really ends up packaged.

```
$ rpm -qpil RPMS/PackageKit-1.2.5-1.aarch64.rpm |grep dbus-1
/etc/dbus-1/system.d/org.freedesktop.PackageKit.conf
/usr/share/dbus-1/interfaces/org.freedesktop.PackageKit.Transaction.xml
/usr/share/dbus-1/interfaces/org.freedesktop.PackageKit.xml
/usr/share/dbus-1/system-services/org.freedesktop.PackageKit.service
```

**Review by Thaodan — APPROVED** [review]

**Pull request merged**